Sources/BraintreePayPal/BTPayPalError.swift

```diff
@@ -112,6 +117,8 @@ public enum BTPayPalError: Error, CustomNSError, LocalizedError, Equatable {
             return "Missing BA Token for PayPal App Switch."
         case .missingPayPalRequest:
             return "The PayPal Request was missing or invalid."
+        case .missingRiskCorrelationID:
```

Comment:
Lets confirm with product that we should fail if we cannot fetch a `riskCorrelationID`. If so, we should throw this error before making the call to `generate_edit_fi_url`

Reply:
If we are guaranteeing the value of `riskCorrelationID` on line 489 we shouldn't need to use the optional `clientMetadataID` at all. Instead we can just pass the `riskCorrelationID` into the `handlePayPalEditFIRequest`, `performEditSwitchRequest`, and `performEditSwitchRequest` methods. While we don't expect this to ever happen if for some reason it does it'd be odd to throw this error so late in the flow. If we instead do the above we can remove the error entirely.

Reply:
Do we need clientMetadataID for analytics for edit flow?

Reply:
You are right. It is not right to have public error for merchants for non-existent scenario.

Reply:
Yes, I think we still need the different parameters functions because one uses its request property (sending in the raw riskCorrelationID sent in by merchant in error handling flow) and one generated internally by us in regular edit flow (requiring argument in parameters function)